src/librustc/ty/util.rs

```diff
@@ -402,17 +402,20 @@ impl<'a, 'gcx, 'tcx> TyCtxt<'a, 'gcx, 'tcx> {
 // `isize` completely when hashing. To ensure that these don't leak in we use a
 // custom hasher implementation here which inflates the size of these to a `u64`
 // and `i64`.
-struct WidenUsizeHasher<H> {
+//
+// The same goes for endianess: We always convert multi-byte integers to little
+// endian before hashing.
```

Comment:
Can we turn this into a doc comment? Maybe even make it public, seems useful elsewhere.